packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -180,7 +180,17 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
         return bailoutOnAlreadyFinishedWork(current, workInProgress);
       }
     }
-    const nextChildren = render(nextProps, ref);
+
+    let nextChildren;
+    if (__DEV__) {
+      ReactCurrentOwner.current = workInProgress;
```

Comment:
This doesn't look right. We shouldn't set current owner, which is runtime behavior, only in DEV. This could mess up certain new features in prod for example.

Reply:
We were supposed to use the debug frame thing for DEV warnings instead of owner so that we didn't have to do this. Perhaps the fix is to change how the context warning works so that it doesn't rely on current owner? @acdlite

Reply:
oh hm. for functional and indeterminate components it looks DEV-only:

https://github.com/facebook/react/blob/e8394daaba08518f6aea2fc127fe505982a493ba/packages/react-reconciler/src/ReactFiberBeginWork.js#L260-L344

I guess that's an oversight?

Reply:
I think at the time this was intentional because we didn't separate those concepts, and didn't want to have the overhead of setting a property in PROD when it doesn't end up being used (functional components can't be ref owners which was the only runtime behavior of this feature).

Reply:
We just have to be better at following through when we change strategies like introducing debug frame.